packages/mongodb-chatbot-ui/src/MessageRating.tsx

```diff
@@ -0,0 +1,160 @@
+import { css } from "@emotion/css";
+import { Body } from "@leafygreen-ui/typography";
+import {
+  MessageRating,
+  type MessageRatingProps,
+} from "@lg-chat/message-rating";
+import { useRef, useState } from "react";
+import { InlineMessageFeedback } from "@lg-chat/message-feedback";
+// @ts-expect-error Typescript imports of icons currently not supported
+import WarningIcon from "@leafygreen-ui/icon/dist/Warning";
+import { spacing } from "@leafygreen-ui/tokens";
+import { palette } from "@leafygreen-ui/palette";
+import { useDarkMode } from "@leafygreen-ui/leafygreen-provider";
+import { CharacterCount } from "./InputBar";
+
+export type RatingCommentStatus = "none" | "submitted" | "abandoned";
+
+export type MessageRatingWithFeedbackCommentProps = {
+  submit: (commentText: string) => void | Promise<void>;
+  abandon: () => void;
+  status: RatingCommentStatus;
+  errorMessage?: string;
+  clearErrorMessage?: () => void;
+  maxCommentCharacterCount?: number;
+  messageRatingProps: MessageRatingProps;
+};
+
+export function MessageRatingWithFeedbackComment(
+  props: MessageRatingWithFeedbackCommentProps
+) {
+  const {
+    submit,
+    abandon,
+    status,
+    errorMessage,
+    maxCommentCharacterCount,
+    messageRatingProps,
+  } = props;
+
+  const hasRating = messageRatingProps.value !== undefined;
+
+  // TODO: Use this to animate the transition after https://jira.mongodb.org/browse/LG-3965 is merged.
+  // const ratingCommentInputVisible = hasRating && status !== "submitted" && status !== "abandoned";
```

Comment:
should we have ticket for this?

Reply:
Sure https://jira.mongodb.org/browse/EAI-618